ractor_actors/src/net/tcp/listener.rs

```diff
@@ -3,98 +3,88 @@
 // This source code is licensed under both the MIT license found in the
 // LICENSE-MIT file in the root directory of this source tree.
 
-//! Tcp listener actor
+//! TCP Server to accept incoming sessions
 
-use std::marker::PhantomData;
-
-use ractor::ActorProcessingErr;
-use ractor::{Actor, ActorRef};
+use ractor::{Actor, ActorProcessingErr, ActorRef};
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::Arc;
 use tokio::net::TcpListener;
 
-use super::{IncomingEncryptionMode, SessionAcceptor};
+use super::{IncomingEncryptionMode, NetworkStream};
 
-/// A Tcp Socket [Listener] responsible for accepting new connections.
-pub struct Listener<R>
-where
-    R: SessionAcceptor,
-{
-    _r: PhantomData<fn() -> R>,
-}
+/// A Tcp Socket [Listener] responsible for creating the socket and accepting new connections. When
+/// a client connects, the `on_connection` will be called. The callback should create a new
+/// [super::session::Session]s which handle the message sending and receiving over the socket.
+///
+/// The [Listener] supervises all the TCP [super::session::Session] actors and is responsible for
+/// logging connects and disconnects.
+pub struct Listener {
+    port: super::NetworkPort,
```

Comment:
actors, ideally, shouldn't be constructed with arguments. That's why the actor startup routines have an Arguments type. This is because the actor only has access to `&self` and can't access owned instances of any of the provided values, nor can it mutate them if necessary. Ideally the "actor" type is just a stub struct, and the arguments & state contain all necessary owned logic.